### PR TITLE
Update Variance Reason Options (and add deprecations)

### PR DESCRIPTION
**Pull request by wes-cutting**
To be considered with [Cord-Field PR](https://github.com/SeedCompany/cord-field/pull/1486)
[Ticket explaining changes](https://seed-company-squad.monday.com/boards/3451697530/pulses/5123136460)

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/5320368981) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by CarsonF — CHANGES_REQUESTED** [review]
Noting that we talked about adding a `deprecated` field

File: src/components/progress-report/variance-explanation/variance-explanation.dto.ts [hunk 1]
```diff
@@ -26,6 +26,9 @@ export abstract class ProgressReportVarianceExplanation {
 
   readonly report: ProgressReport & SetUnsecuredType<ID>;
 
+  @Field()
+  readonly deprecated: SecuredStringList;
```
Comment (CarsonF) [inline src/components/progress-report/variance-explanation/variance-explanation.dto.ts:30]: This isn't needed. The other one is sufficient

Comment (CarsonF) [reply src/components/progress-report/variance-explanation/variance-explanation.dto.ts:30]: [Here](https://github.com/SeedCompany/cord-api-v3/pull/2914/files#diff-3c01125b0bcbed569795854d33632f57f8f35e2dac24bd6090c2ee8015376b94R19). This DTO shape is for the actual saved value for each report, we don't need to track the options here or if they are deprecated.

Comment (wes-cutting) [reply src/components/progress-report/variance-explanation/variance-explanation.dto.ts:30]: Oh I see

File: src/components/progress-report/variance-explanation/variance-explanation.repository.ts [hunk 2]
```diff
@@ -39,6 +39,7 @@ export class ProgressReportVarianceExplanationRepository extends DtoRepository(
   protected hydrate() {
     const defaults: UnsecuredDto<VarianceExplanation> & ExpressionInput = {
       report: 'report.id' as ID,
+      deprecated: [],
```
Comment (CarsonF) [inline src/components/progress-report/variance-explanation/variance-explanation.repository.ts:42]: Again not needed. This deprecated list is only for the options, not for actual saved values for reports.

**Review by CarsonF — APPROVED** [review]
Thanks @wes-cutting!

I fixed a few things:
- I forgot to import the migration into the nestjs module.
- I moved the note about the deprecated list into the description field, so it comes across in the GQL schema.
- I also added a check to ensure that deprecated reasons cannot be selected (if they are not currently selected).

**Pull request merged**